tests/rptest/tests/offset_for_leader_epoch_archival_test.py

```diff
@@ -86,9 +85,14 @@ def test_querying_remote_partitions(self, remote_reads):
         epoch_offsets = {}
         rpk = RpkTool(self.redpanda)
         self.client().create_topic(topic)
+        remote_reads_str = 'true' if remote_reads else 'false'
         rpk.alter_topic_config(topic.name, "redpanda.remote.read",
-                               str(remote_reads))
+                               remote_reads_str)
```

Comment:
This was actually a test bug previously!

[`parse_and_set_shadow_indexing_mode()` has a default_match case](https://github.com/redpanda-data/redpanda/blob/4c2c24c468048905a00d4f52bc546e97ad88bcbf/src/v/kafka/server/handlers/alter_configs.cc#L55) which silently disables the value, and Python `True` does not match any of the other `string_switch` cases. 

I.e, this NEVER properly worked in setting the tiered storage permission to `True`, and with the fixes in this PR, [`parse_and_set_bool()`](https://github.com/redpanda-data/redpanda/blob/4c2c24c468048905a00d4f52bc546e97ad88bcbf/src/v/kafka/server/handlers/configs/config_utils.h#L637-L639) actually hits this `boost::bad_lexical_cast` outcome.

The only acceptable values are string `'true'` and string `'false'`.

Reply:
Nice find!

I think it's worth making that switch more robust though by ignoring the case. That's what Kafka seems to be doing as well, so it makes sense to match that: https://github.com/apache/kafka/blob/818ee8a581b92ccd0434310ea3b108e76daea565/clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java#L690-L710

"yes" and "no" doesn't seem to be handled in Kafka but it might be worth porting that over from `parse_and_set_shadow_indexing_mode` just to be safe.

I would expect that most clients just use "true"/"false", but it seems easy to be more robust here to various inputs. Throwing on no match definitely seems like the right thing to do, though.

Reply:
Added a `std::transform()` call in `parse_and_set_bool()` to ignore case.